### PR TITLE
Implement: Post request with error or success message and save to dat…

### PR DESCRIPTION
**Pull request by stevethomas**
…abase

### CODE REVIEW

File: app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Post;
```
Comment (jabui) [inline app/Http/Controllers/PostController.php]: Order imports by ascending string length - smallest to largest.

File: app/Http/Controllers/PostController.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class PostController extends Controller
+{
+    public function store(Request $request)
+    {
+
```
Comment (jabui) [inline app/Http/Controllers/PostController.php]: Don't keep any unnecessary whitespace

File: app/Http/Controllers/PostController.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Post;
+
+class PostController extends Controller
+{
+    public function store(Request $request)
```
Comment (jabui) [inline app/Http/Controllers/PostController.php]: Doc block each method, here's an example - http://docs.phpdoc.org/getting-started/your-first-set-of-documentation.html#example

File: app/Http/Controllers/PostController.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {
+
+      $validatedData = $request->validate([
+          'name' => 'required',
+          'description' => 'required'
+      ]);
+
+      $user = auth()->user();
```
Comment (jabui) [inline app/Http/Controllers/PostController.php]: Grab the user from the request instead `$request->user()`

File: database/migrations/2018_08_08_102805_create_post_table.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('posts', function (Blueprint $table) {
+          $table->increments('id');
+          $table->integer('user_id')->nullable();
+          $table->string('name')->nullable();
+          $table->text('description')->nullable();
+          $table->timestamps();
```
Comment (stevethomas) [inline database/migrations/2018_08_08_102805_create_post_table.php]: If the `user_id`, `name` and `description` are all required fields via authentication or validation, then we would not make them `nullable()` as well.

File: resources/views/home.blade.php [hunk 6]
```diff
@@ -18,6 +18,45 @@
                 </div>
             </div>
         </div>
+        <div class="col-md-8">
+          <br>
+          <form method="post" action="{{url('add_post')}}" enctype="multipart/form-data">
```
Comment (stevethomas) [inline resources/views/home.blade.php]: There is no need to use enctype attribute on this form; this is only used when the form supports file attachments.

**Comment by jeffersonbabuyo** [comment]
Good day Steve,

I just push my code updates base on the code review. Thanks!

Jefferson Babuyo
Senior Backend Developer/DevOps

File: app/Http/Controllers/PostController.php (hunk 2 above)
Comment (jeffersonbabuyo) [reply app/Http/Controllers/PostController.php]: whitespace remove, installed editorconfig plugin in my atom so .editorconfig will work.

File: app/Http/Controllers/PostController.php (hunk 4 above)
Comment (jeffersonbabuyo) [reply app/Http/Controllers/PostController.php]: Agree, we can actually do this: `$post->user_id = $request->user()->id;` to minimize variable declaration.

File: database/migrations/2018_08_08_102805_create_post_table.php (hunk 5 above)
Comment (jeffersonbabuyo) [reply database/migrations/2018_08_08_102805_create_post_table.php]: Thanks @stevethomas , I forgot to remove `nullable()` because I implemented the form validation late when I read the bonus points

File: resources/views/home.blade.php (hunk 6 above)
Comment (jeffersonbabuyo) [reply resources/views/home.blade.php]: Thanks, enctype attribute is now remove
